### PR TITLE
change ec2 dnsSuffix principal in policy

### PR DESCRIPTION
**Pull request by liutao-samsung**
### Proposed changes

IAM principal in policy may be different in different aws partition.
https://github.com/weaveworks/eksctl/blob/main/pkg/cfn/builder/partition.go#L9

### CODE REVIEW

File: nodejs/eks/cluster.ts [hunk 1]
```diff
@@ -401,6 +401,7 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     };
 
     const partition = pulumi.output(aws.getPartition({ provider })).partition;
+    const dnsSuffix = pulumi.output(aws.getPartition({ provider })).dnsSuffix;
```
Comment (danielrbradley) [inline nodejs/eks/cluster.ts:404]: We should be able to fetch both in a single call and grab the two values we need.

```suggestion
    const { partition, dnsSuffix } = pulumi.output(aws.getPartition({ provider }));
```

**Review by danielrbradley — CHANGES_REQUESTED** [review]
This looks great thanks! 

Just a minor tweak needed to avoid us doing two roundtrips to get the partition information then we'll get this merged

**Comment by liutao-samsung** [comment]
What should I do next?

**Comment by liutao-samsung** [comment]
@danielrbradley  Hi ~ , would you kick off  the test again ?

**Comment by danielrbradley** [comment]
I'm afraid we're still getting test failures, though it might still be unrelated to your change.

```
pulumi:pulumi:Stack (example-cluster-p-it-fv-az419-8-cluster-5c222dca):
      error: Running program '/tmp/p-it-fv-az419-8-cluster-5c222dca-2770179391' failed with an unhandled exception:
      TypeError: Cannot read properties of undefined (reading 'data')
          at /home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/cluster.ts:586:103
          at /home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/@pulumi/output.ts:383:31
          at Generator.next (<anonymous>)
          at /home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/@pulumi/pulumi/output.js:21:71
          at new Promise (<anonymous>)
          at __awaiter (/home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/@pulumi/pulumi/output.js:17:12)
          at applyHelperAsync (/home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/@pulumi/pulumi/output.js:236:12)
          at /home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/@pulumi/output.ts:302:65
      error: update failed
```

A brief glance looks like it's related to `clusterCertificateAuthority`. This will probably required a bit more investigation before we get a clean build. Sorry for the slow process. 

cc: @roothorp

**Comment by roothorp** [comment]
I think this is the issue from #676?

**Review by danielrbradley — APPROVED** [review]

**Pull request merged**